planning/behavior_velocity_planner/src/utilization/util.cpp

```diff
@@ -244,6 +244,86 @@ double calcJudgeLineDistWithJerkLimit(
   return std::max(0.0, x1 + x2 + x3);
 }
 
+double findReachTime(
+  const double j, const double a, const double v, const double d, const double min,
+  const double max)
+{
+  auto f = [](const double t, const double j, const double a, const double v, const double d) {
+    return j * t * t * t / 6.0 + a * t * t / 2.0 + v * t - d;
+  };
+  if (f(min, j, a, v, d) > 0 || f(max, j, a, v, d) < 0) {
+    std::logic_error("[behavior_velocity](findReachTime): search range is invalid");
+  }
+  const double eps = 1e-5;
+  const int warn_iter = 100;
+  double lower = min;
+  double upper = max;
+  double t;
+  int iter = 0;
+  for (int i = 0;; i++) {
+    t = 0.5 * (lower + upper);
+    const double fx = f(t, j, a, v, d);
+    // std::cout<<"fx: "<<fx<<" up: "<<upper<<" lo: "<<lower<<" t: "<<t<<std::endl;
+    if (std::abs(fx) < eps) {
+      break;
+    } else if (fx > 0.0) {
+      upper = t;
+    } else {
+      lower = t;
+    }
+    iter++;
+    if (iter > warn_iter)
+      std::cerr << "[behavior_velocity](findReachTime): current iter is over warning" << std::endl;
+  }
+  // std::cout<<"iter: "<<iter<<std::endl;
+  return t;
+}
+
+double calcMaxSlowDownVelocityFromDistanceToTarget(
```

Comment:
This function calculates the speed at target point when the vehicle deceleates with max_slowdown_jerk/max_slowdown_accel.

So, I think this function name should be, for example, calcDecelerationVelocityInTargetPoint

Reply:
(Is this maximum slow down velocity? isn't it minimum?)

Reply:
[7a8d695](https://github.com/autowarefoundation/autoware.universe/pull/388/commits/7a8d69553edbacf057b51314b7f214777e5ce5ab)